### ISSUE TITLE
[BUG] Unicode handling question mark placeholder character

### ISSUE THREAD

**Issue by munvoseli** [mawww/kakoune#4955]
### Version of Kakoune

v2022.10.31

### Reproducer

Have 0xC3 0xB6 (UTF-8 for Unicode character ö with codepoint 00F6) in a file. (this also happens with other Unicode characters, such as ä)

Use st or xterm to launch Kakoune.

### Outcome

![2023-08-04-094127_170x44_scrot](https://github.com/mawww/kakoune/assets/60828599/15e57aa4-42c8-48c3-ba50-07025577ba8e)

### Expectations

![2023-08-04-093955_164x50_scrot](https://github.com/mawww/kakoune/assets/60828599/af93017f-0efe-4b7e-8137-0007be4fb281)

This is the expected behavior, and occurs with head+st, head+xterm, less+st, less+xterm, nano+st, nano+xterm, vim+xt, and vim+xterm.

### Additional information

I have not tested terminals other than st and xterm.

This is the output of `locale`:
```
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

**Comment by krobelus** [comment]
I can't reproduce with this file:

```
$ xxd foo
00000000: 6973 7420 6c69 6d61 6bc3 b620 2278 0a    ist limak.. "x.
```

**Comment by munvoseli** [comment]
Sorry.

It appears to be fixed in the v2023.07.29, but not v2022.10.31 (from nix).
